// @google-shopping/accounts/build/src/index.js
"use strict";
// Copyright 2026 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by synthtool. **
// ** https://github.com/googleapis/synthtool **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.protos = exports.UserServiceClient = exports.TermsOfServiceServiceClient = exports.TermsOfServiceAgreementStateServiceClient = exports.ShippingSettingsServiceClient = exports.RegionsServiceClient = exports.ProgramsServiceClient = exports.OnlineReturnPolicyServiceClient = exports.OmnichannelSettingsServiceClient = exports.LfpProvidersServiceClient = exports.HomepageServiceClient = exports.GbpAccountsServiceClient = exports.EmailPreferencesServiceClient = exports.DeveloperRegistrationServiceClient = exports.CheckoutSettingsServiceClient = exports.BusinessInfoServiceClient = exports.BusinessIdentityServiceClient = exports.AutomaticImprovementsServiceClient = exports.AutofeedSettingsServiceClient = exports.AccountsServiceClient = exports.AccountServicesServiceClient = exports.AccountRelationshipsServiceClient = exports.AccountIssueServiceClient = exports.v1beta = exports.v1 = void 0;
const v1 = require("./v1");
exports.v1 = v1;
const v1beta = require("./v1beta");
exports.v1beta = v1beta;
const AccountIssueServiceClient = v1.AccountIssueServiceClient;
exports.AccountIssueServiceClient = AccountIssueServiceClient;
const AccountRelationshipsServiceClient = v1.AccountRelationshipsServiceClient;
exports.AccountRelationshipsServiceClient = AccountRelationshipsServiceClient;
const AccountServicesServiceClient = v1.AccountServicesServiceClient;
exports.AccountServicesServiceClient = AccountServicesServiceClient;
const AccountsServiceClient = v1.AccountsServiceClient;
exports.AccountsServiceClient = AccountsServiceClient;
const AutofeedSettingsServiceClient = v1.AutofeedSettingsServiceClient;
exports.AutofeedSettingsServiceClient = AutofeedSettingsServiceClient;
const AutomaticImprovementsServiceClient = v1.AutomaticImprovementsServiceClient;
exports.AutomaticImprovementsServiceClient = AutomaticImprovementsServiceClient;
const BusinessIdentityServiceClient = v1.BusinessIdentityServiceClient;
exports.BusinessIdentityServiceClient = BusinessIdentityServiceClient;
const BusinessInfoServiceClient = v1.BusinessInfoServiceClient;
exports.BusinessInfoServiceClient = BusinessInfoServiceClient;
const CheckoutSettingsServiceClient = v1.CheckoutSettingsServiceClient;
exports.CheckoutSettingsServiceClient = CheckoutSettingsServiceClient;
const DeveloperRegistrationServiceClient = v1.DeveloperRegistrationServiceClient;
exports.DeveloperRegistrationServiceClient = DeveloperRegistrationServiceClient;
const EmailPreferencesServiceClient = v1.EmailPreferencesServiceClient;
exports.EmailPreferencesServiceClient = EmailPreferencesServiceClient;
const GbpAccountsServiceClient = v1.GbpAccountsServiceClient;
exports.GbpAccountsServiceClient = GbpAccountsServiceClient;
const HomepageServiceClient = v1.HomepageServiceClient;
exports.HomepageServiceClient = HomepageServiceClient;
const LfpProvidersServiceClient = v1.LfpProvidersServiceClient;
exports.LfpProvidersServiceClient = LfpProvidersServiceClient;
const OmnichannelSettingsServiceClient = v1.OmnichannelSettingsServiceClient;
exports.OmnichannelSettingsServiceClient = OmnichannelSettingsServiceClient;
const OnlineReturnPolicyServiceClient = v1.OnlineReturnPolicyServiceClient;
exports.OnlineReturnPolicyServiceClient = OnlineReturnPolicyServiceClient;
const ProgramsServiceClient = v1.ProgramsServiceClient;
exports.ProgramsServiceClient = ProgramsServiceClient;
const RegionsServiceClient = v1.RegionsServiceClient;
exports.RegionsServiceClient = RegionsServiceClient;
const ShippingSettingsServiceClient = v1.ShippingSettingsServiceClient;
exports.ShippingSettingsServiceClient = ShippingSettingsServiceClient;
const TermsOfServiceAgreementStateServiceClient = v1.TermsOfServiceAgreementStateServiceClient;
exports.TermsOfServiceAgreementStateServiceClient = TermsOfServiceAgreementStateServiceClient;
const TermsOfServiceServiceClient = v1.TermsOfServiceServiceClient;
exports.TermsOfServiceServiceClient = TermsOfServiceServiceClient;
const UserServiceClient = v1.UserServiceClient;
exports.UserServiceClient = UserServiceClient;
exports.default = { v1, v1beta, AccountIssueServiceClient, AccountRelationshipsServiceClient, AccountServicesServiceClient, AccountsServiceClient, AutofeedSettingsServiceClient, AutomaticImprovementsServiceClient, BusinessIdentityServiceClient, BusinessInfoServiceClient, CheckoutSettingsServiceClient, DeveloperRegistrationServiceClient, EmailPreferencesServiceClient, GbpAccountsServiceClient, HomepageServiceClient, LfpProvidersServiceClient, OmnichannelSettingsServiceClient, OnlineReturnPolicyServiceClient, ProgramsServiceClient, RegionsServiceClient, ShippingSettingsServiceClient, TermsOfServiceAgreementStateServiceClient, TermsOfServiceServiceClient, UserServiceClient };
const protos = require("../protos/protos");
exports.protos = protos;
