// @google-shopping/accounts/build/src/v1beta/email_preferences_service_client.js
"use strict";
// Copyright 2026 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailPreferencesServiceClient = void 0;
const jsonProtos = require("../../protos/protos.json");
const google_gax_1 = require("google-gax");
/**
 * Client JSON configuration object, loaded from
 * `src/v1beta/email_preferences_service_client_config.json`.
 * This file defines retry strategy and timeouts for all API methods in this library.
 */
const gapicConfig = require("./email_preferences_service_client_config.json");
const version = require('../../../package.json').version;
/**
 *  Service to support the `EmailPreferences` API.
 *
 *  This service only permits retrieving and updating email preferences for the
 *  authenticated user.
 * @class
 * @memberof v1beta
 */
class EmailPreferencesServiceClient {
    _terminated = false;
    _opts;
    _providedCustomServicePath;
    _gaxModule;
    _gaxGrpc;
    _protos;
    _defaults;
    _universeDomain;
    _servicePath;
    _log = google_gax_1.loggingUtils.log('accounts');
    auth;
    descriptors = {
        page: {},
        stream: {},
        longrunning: {},
        batching: {},
    };
    warn;
    innerApiCalls;
    pathTemplates;
    emailPreferencesServiceStub;
    /**
     * Construct an instance of EmailPreferencesServiceClient.
     *
     * @param {object} [options] - The configuration object.
     * The options accepted by the constructor are described in detail
     * in [this document](https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#creating-the-client-instance).
     * The common options are:
     * @param {object} [options.credentials] - Credentials object.
     * @param {string} [options.credentials.client_email]
     * @param {string} [options.credentials.private_key]
     * @param {string} [options.email] - Account email address. Required when
     *     using a .pem or .p12 keyFilename.
     * @param {string} [options.keyFilename] - Full path to the a .json, .pem, or
     *     .p12 key downloaded from the Google Developers Console. If you provide
     *     a path to a JSON file, the projectId option below is not necessary.
     *     NOTE: .pem and .p12 require you to specify options.email as well.
     * @param {number} [options.port] - The port on which to connect to
     *     the remote host.
     * @param {string} [options.projectId] - The project ID from the Google
     *     Developer's Console, e.g. 'grape-spaceship-123'. We will also check
     *     the environment variable GCLOUD_PROJECT for your project ID. If your
     *     app is running in an environment which supports
     *     {@link https://cloud.google.com/docs/authentication/application-default-credentials Application Default Credentials},
     *     your project ID will be detected automatically.
     * @param {string} [options.apiEndpoint] - The domain name of the
     *     API remote host.
     * @param {gax.ClientConfig} [options.clientConfig] - Client configuration override.
     *     Follows the structure of {@link gapicConfig}.
     * @param {boolean} [options.fallback] - Use HTTP/1.1 REST mode.
     *     For more information, please check the
     *     {@link https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#http11-rest-api-mode documentation}.
     * @param {gax} [gaxInstance]: loaded instance of `google-gax`. Useful if you
     *     need to avoid loading the default gRPC version and want to use the fallback
     *     HTTP implementation. Load only fallback version and pass it to the constructor:
     *     ```
     *     const gax = require('google-gax/build/src/fallback'); // avoids loading google-gax with gRPC
     *     const client = new EmailPreferencesServiceClient({fallback: true}, gax);
     *     ```
     */
    constructor(opts, gaxInstance) {
        // Ensure that options include all the required fields.
        const staticMembers = this.constructor;
        if (opts?.universe_domain && opts?.universeDomain && opts?.universe_domain !== opts?.universeDomain) {
            throw new Error('Please set either universe_domain or universeDomain, but not both.');
        }
        const universeDomainEnvVar = (typeof process === 'object' && typeof process.env === 'object') ? process.env['GOOGLE_CLOUD_UNIVERSE_DOMAIN'] : undefined;
        this._universeDomain = opts?.universeDomain ?? opts?.universe_domain ?? universeDomainEnvVar ?? 'googleapis.com';
        this._servicePath = 'merchantapi.' + this._universeDomain;
        const servicePath = opts?.servicePath || opts?.apiEndpoint || this._servicePath;
        this._providedCustomServicePath = !!(opts?.servicePath || opts?.apiEndpoint);
        const port = opts?.port || staticMembers.port;
        const clientConfig = opts?.clientConfig ?? {};
        const fallback = opts?.fallback ?? (typeof window !== 'undefined' && typeof window?.fetch === 'function');
        opts = Object.assign({ servicePath, port, clientConfig, fallback }, opts);
        // Request numeric enum values if REST transport is used.
        opts.numericEnums = true;
        // If scopes are unset in options and we're connecting to a non-default endpoint, set scopes just in case.
        if (servicePath !== this._servicePath && !('scopes' in opts)) {
            opts['scopes'] = staticMembers.scopes;
        }
        // Load google-gax module synchronously if needed
        if (!gaxInstance) {
            gaxInstance = require('google-gax');
        }
        // Choose either gRPC or proto-over-HTTP implementation of google-gax.
        this._gaxModule = opts.fallback ? gaxInstance.fallback : gaxInstance;
        // Create a `gaxGrpc` object, with any grpc-specific options sent to the client.
        this._gaxGrpc = new this._gaxModule.GrpcClient(opts);
        // Save options to use in initialize() method.
        this._opts = opts;
        // Save the auth object to the client, for use by other methods.
        this.auth = this._gaxGrpc.auth;
        // Set useJWTAccessWithScope on the auth object.
        this.auth.useJWTAccessWithScope = true;
        // Set defaultServicePath on the auth object.
        this.auth.defaultServicePath = this._servicePath;
        // Set the default scopes in auth client if needed.
        if (servicePath === this._servicePath) {
            this.auth.defaultScopes = staticMembers.scopes;
        }
        // Determine the client header string.
        const clientHeader = [
            `gax/${this._gaxModule.version}`,
            `gapic/${version}`,
        ];
        if (typeof process === 'object' && 'versions' in process) {
            clientHeader.push(`gl-node/${process.versions.node}`);
        }
        else {
            clientHeader.push(`gl-web/${this._gaxModule.version}`);
        }
        if (!opts.fallback) {
            clientHeader.push(`grpc/${this._gaxGrpc.grpcVersion}`);
        }
        else {
            clientHeader.push(`rest/${this._gaxGrpc.grpcVersion}`);
        }
        if (opts.libName && opts.libVersion) {
            clientHeader.push(`${opts.libName}/${opts.libVersion}`);
        }
        // Load the applicable protos.
        this._protos = this._gaxGrpc.loadProtoJSON(jsonProtos);
        // This API contains "path templates"; forward-slash-separated
        // identifiers to uniquely identify resources within the API.
        // Create useful helper objects for these.
        this.pathTemplates = {
            accountPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}'),
            accountIssuePathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/issues/{issue}'),
            accountTaxPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/accounttax/{tax}'),
            autofeedSettingsPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/autofeedSettings'),
            automaticImprovementsPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/automaticImprovements'),
            businessIdentityPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/businessIdentity'),
            businessInfoPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/businessInfo'),
            checkoutSettingsPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/programs/{program}/checkoutSettings'),
            emailPreferencesPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/users/{email}/emailPreferences'),
            gbpAccountPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/gbpAccounts/{gbp_account}'),
            homepagePathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/homepage'),
            lfpProviderPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/omnichannelSettings/{omnichannel_setting}/lfpProviders/{lfp_provider}'),
            omnichannelSettingPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/omnichannelSettings/{omnichannel_setting}'),
            onlineReturnPolicyPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/onlineReturnPolicies/{return_policy}'),
            programPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/programs/{program}'),
            regionPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/regions/{region}'),
            shippingSettingsPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/shippingSettings'),
            termsOfServicePathTemplate: new this._gaxModule.PathTemplate('termsOfService/{version}'),
            termsOfServiceAgreementStatePathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/termsOfServiceAgreementStates/{identifier}'),
            userPathTemplate: new this._gaxModule.PathTemplate('accounts/{account}/users/{email}'),
        };
        // Put together the default options sent with requests.
        this._defaults = this._gaxGrpc.constructSettings('google.shopping.merchant.accounts.v1beta.EmailPreferencesService', gapicConfig, opts.clientConfig || {}, { 'x-goog-api-client': clientHeader.join(' ') });
        // Set up a dictionary of "inner API calls"; the core implementation
        // of calling the API is handled in `google-gax`, with this code
        // merely providing the destination and request information.
        this.innerApiCalls = {};
        // Add a warn function to the client constructor so it can be easily tested.
        this.warn = this._gaxModule.warn;
    }
    /**
     * Initialize the client.
     * Performs asynchronous operations (such as authentication) and prepares the client.
     * This function will be called automatically when any class method is called for the
     * first time, but if you need to initialize it before calling an actual method,
     * feel free to call initialize() directly.
     *
     * You can await on this method if you want to make sure the client is initialized.
     *
     * @returns {Promise} A promise that resolves to an authenticated service stub.
     */
    initialize() {
        // If the client stub promise is already initialized, return immediately.
        if (this.emailPreferencesServiceStub) {
            return this.emailPreferencesServiceStub;
        }
        // Put together the "service stub" for
        // google.shopping.merchant.accounts.v1beta.EmailPreferencesService.
        this.emailPreferencesServiceStub = this._gaxGrpc.createStub(this._opts.fallback ?
            this._protos.lookupService('google.shopping.merchant.accounts.v1beta.EmailPreferencesService') :
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            this._protos.google.shopping.merchant.accounts.v1beta.EmailPreferencesService, this._opts, this._providedCustomServicePath);
        // Iterate over each of the methods that the service provides
        // and create an API call method for each.
        const emailPreferencesServiceStubMethods = ['getEmailPreferences', 'updateEmailPreferences'];
        for (const methodName of emailPreferencesServiceStubMethods) {
            const callPromise = this.emailPreferencesServiceStub.then(stub => (...args) => {
                if (this._terminated) {
                    return Promise.reject('The client has already been closed.');
                }
                const func = stub[methodName];
                return func.apply(stub, args);
            }, (err) => () => {
                throw err;
            });
            const descriptor = undefined;
            const apiCall = this._gaxModule.createApiCall(callPromise, this._defaults[methodName], descriptor, this._opts.fallback);
            this.innerApiCalls[methodName] = apiCall;
        }
        return this.emailPreferencesServiceStub;
    }
    /**
     * The DNS address for this API service.
     * @deprecated Use the apiEndpoint method of the client instance.
     * @returns {string} The DNS address for this service.
     */
    static get servicePath() {
        if (typeof process === 'object' && typeof process.emitWarning === 'function') {
            process.emitWarning('Static servicePath is deprecated, please use the instance method instead.', 'DeprecationWarning');
        }
        return 'merchantapi.googleapis.com';
    }
    /**
     * The DNS address for this API service - same as servicePath.
     * @deprecated Use the apiEndpoint method of the client instance.
     * @returns {string} The DNS address for this service.
     */
    static get apiEndpoint() {
        if (typeof process === 'object' && typeof process.emitWarning === 'function') {
            process.emitWarning('Static apiEndpoint is deprecated, please use the instance method instead.', 'DeprecationWarning');
        }
        return 'merchantapi.googleapis.com';
    }
    /**
     * The DNS address for this API service.
     * @returns {string} The DNS address for this service.
     */
    get apiEndpoint() {
        return this._servicePath;
    }
    get universeDomain() {
        return this._universeDomain;
    }
    /**
     * The port for this API service.
     * @returns {number} The default port for this service.
     */
    static get port() {
        return 443;
    }
    /**
     * The scopes needed to make gRPC calls for every method defined
     * in this service.
     * @returns {string[]} List of default scopes.
     */
    static get scopes() {
        return [
            'https://www.googleapis.com/auth/content'
        ];
    }
    /**
     * Return the project ID used by this class.
     * @returns {Promise} A promise that resolves to string containing the project ID.
     */
    getProjectId(callback) {
        if (callback) {
            this.auth.getProjectId(callback);
            return;
        }
        return this.auth.getProjectId();
    }
    getEmailPreferences(request, optionsOrCallback, callback) {
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] = this._gaxModule.routingHeader.fromParams({
            'name': request.name ?? '',
        });
        this.initialize().catch(err => { throw err; });
        this._log.info('getEmailPreferences request %j', request);
        const wrappedCallback = callback
            ? (error, response, options, rawResponse) => {
                this._log.info('getEmailPreferences response %j', response);
                callback(error, response, options, rawResponse); // We verified callback above.
            }
            : undefined;
        return this.innerApiCalls.getEmailPreferences(request, options, wrappedCallback)
            ?.then(([response, options, rawResponse]) => {
            this._log.info('getEmailPreferences response %j', response);
            return [response, options, rawResponse];
        }).catch((error) => {
            if (error && 'statusDetails' in error && error.statusDetails instanceof Array) {
                const protos = this._gaxModule.protobuf.Root.fromJSON(jsonProtos);
                error.statusDetails = (0, google_gax_1.decodeAnyProtosInArray)(error.statusDetails, protos);
            }
            throw error;
        });
    }
    updateEmailPreferences(request, optionsOrCallback, callback) {
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] = this._gaxModule.routingHeader.fromParams({
            'email_preferences.name': request.emailPreferences.name ?? '',
        });
        this.initialize().catch(err => { throw err; });
        this._log.info('updateEmailPreferences request %j', request);
        const wrappedCallback = callback
            ? (error, response, options, rawResponse) => {
                this._log.info('updateEmailPreferences response %j', response);
                callback(error, response, options, rawResponse); // We verified callback above.
            }
            : undefined;
        return this.innerApiCalls.updateEmailPreferences(request, options, wrappedCallback)
            ?.then(([response, options, rawResponse]) => {
            this._log.info('updateEmailPreferences response %j', response);
            return [response, options, rawResponse];
        }).catch((error) => {
            if (error && 'statusDetails' in error && error.statusDetails instanceof Array) {
                const protos = this._gaxModule.protobuf.Root.fromJSON(jsonProtos);
                error.statusDetails = (0, google_gax_1.decodeAnyProtosInArray)(error.statusDetails, protos);
            }
            throw error;
        });
    }
    // --------------------
    // -- Path templates --
    // --------------------
    /**
     * Return a fully-qualified account resource name string.
     *
     * @param {string} account
     * @returns {string} Resource name string.
     */
    accountPath(account) {
        return this.pathTemplates.accountPathTemplate.render({
            account: account,
        });
    }
    /**
     * Parse the account from Account resource.
     *
     * @param {string} accountName
     *   A fully-qualified path representing Account resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromAccountName(accountName) {
        return this.pathTemplates.accountPathTemplate.match(accountName).account;
    }
    /**
     * Return a fully-qualified accountIssue resource name string.
     *
     * @param {string} account
     * @param {string} issue
     * @returns {string} Resource name string.
     */
    accountIssuePath(account, issue) {
        return this.pathTemplates.accountIssuePathTemplate.render({
            account: account,
            issue: issue,
        });
    }
    /**
     * Parse the account from AccountIssue resource.
     *
     * @param {string} accountIssueName
     *   A fully-qualified path representing AccountIssue resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromAccountIssueName(accountIssueName) {
        return this.pathTemplates.accountIssuePathTemplate.match(accountIssueName).account;
    }
    /**
     * Parse the issue from AccountIssue resource.
     *
     * @param {string} accountIssueName
     *   A fully-qualified path representing AccountIssue resource.
     * @returns {string} A string representing the issue.
     */
    matchIssueFromAccountIssueName(accountIssueName) {
        return this.pathTemplates.accountIssuePathTemplate.match(accountIssueName).issue;
    }
    /**
     * Return a fully-qualified accountTax resource name string.
     *
     * @param {string} account
     * @param {string} tax
     * @returns {string} Resource name string.
     */
    accountTaxPath(account, tax) {
        return this.pathTemplates.accountTaxPathTemplate.render({
            account: account,
            tax: tax,
        });
    }
    /**
     * Parse the account from AccountTax resource.
     *
     * @param {string} accountTaxName
     *   A fully-qualified path representing AccountTax resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromAccountTaxName(accountTaxName) {
        return this.pathTemplates.accountTaxPathTemplate.match(accountTaxName).account;
    }
    /**
     * Parse the tax from AccountTax resource.
     *
     * @param {string} accountTaxName
     *   A fully-qualified path representing AccountTax resource.
     * @returns {string} A string representing the tax.
     */
    matchTaxFromAccountTaxName(accountTaxName) {
        return this.pathTemplates.accountTaxPathTemplate.match(accountTaxName).tax;
    }
    /**
     * Return a fully-qualified autofeedSettings resource name string.
     *
     * @param {string} account
     * @returns {string} Resource name string.
     */
    autofeedSettingsPath(account) {
        return this.pathTemplates.autofeedSettingsPathTemplate.render({
            account: account,
        });
    }
    /**
     * Parse the account from AutofeedSettings resource.
     *
     * @param {string} autofeedSettingsName
     *   A fully-qualified path representing AutofeedSettings resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromAutofeedSettingsName(autofeedSettingsName) {
        return this.pathTemplates.autofeedSettingsPathTemplate.match(autofeedSettingsName).account;
    }
    /**
     * Return a fully-qualified automaticImprovements resource name string.
     *
     * @param {string} account
     * @returns {string} Resource name string.
     */
    automaticImprovementsPath(account) {
        return this.pathTemplates.automaticImprovementsPathTemplate.render({
            account: account,
        });
    }
    /**
     * Parse the account from AutomaticImprovements resource.
     *
     * @param {string} automaticImprovementsName
     *   A fully-qualified path representing AutomaticImprovements resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromAutomaticImprovementsName(automaticImprovementsName) {
        return this.pathTemplates.automaticImprovementsPathTemplate.match(automaticImprovementsName).account;
    }
    /**
     * Return a fully-qualified businessIdentity resource name string.
     *
     * @param {string} account
     * @returns {string} Resource name string.
     */
    businessIdentityPath(account) {
        return this.pathTemplates.businessIdentityPathTemplate.render({
            account: account,
        });
    }
    /**
     * Parse the account from BusinessIdentity resource.
     *
     * @param {string} businessIdentityName
     *   A fully-qualified path representing BusinessIdentity resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromBusinessIdentityName(businessIdentityName) {
        return this.pathTemplates.businessIdentityPathTemplate.match(businessIdentityName).account;
    }
    /**
     * Return a fully-qualified businessInfo resource name string.
     *
     * @param {string} account
     * @returns {string} Resource name string.
     */
    businessInfoPath(account) {
        return this.pathTemplates.businessInfoPathTemplate.render({
            account: account,
        });
    }
    /**
     * Parse the account from BusinessInfo resource.
     *
     * @param {string} businessInfoName
     *   A fully-qualified path representing BusinessInfo resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromBusinessInfoName(businessInfoName) {
        return this.pathTemplates.businessInfoPathTemplate.match(businessInfoName).account;
    }
    /**
     * Return a fully-qualified checkoutSettings resource name string.
     *
     * @param {string} account
     * @param {string} program
     * @returns {string} Resource name string.
     */
    checkoutSettingsPath(account, program) {
        return this.pathTemplates.checkoutSettingsPathTemplate.render({
            account: account,
            program: program,
        });
    }
    /**
     * Parse the account from CheckoutSettings resource.
     *
     * @param {string} checkoutSettingsName
     *   A fully-qualified path representing CheckoutSettings resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromCheckoutSettingsName(checkoutSettingsName) {
        return this.pathTemplates.checkoutSettingsPathTemplate.match(checkoutSettingsName).account;
    }
    /**
     * Parse the program from CheckoutSettings resource.
     *
     * @param {string} checkoutSettingsName
     *   A fully-qualified path representing CheckoutSettings resource.
     * @returns {string} A string representing the program.
     */
    matchProgramFromCheckoutSettingsName(checkoutSettingsName) {
        return this.pathTemplates.checkoutSettingsPathTemplate.match(checkoutSettingsName).program;
    }
    /**
     * Return a fully-qualified emailPreferences resource name string.
     *
     * @param {string} account
     * @param {string} email
     * @returns {string} Resource name string.
     */
    emailPreferencesPath(account, email) {
        return this.pathTemplates.emailPreferencesPathTemplate.render({
            account: account,
            email: email,
        });
    }
    /**
     * Parse the account from EmailPreferences resource.
     *
     * @param {string} emailPreferencesName
     *   A fully-qualified path representing EmailPreferences resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromEmailPreferencesName(emailPreferencesName) {
        return this.pathTemplates.emailPreferencesPathTemplate.match(emailPreferencesName).account;
    }
    /**
     * Parse the email from EmailPreferences resource.
     *
     * @param {string} emailPreferencesName
     *   A fully-qualified path representing EmailPreferences resource.
     * @returns {string} A string representing the email.
     */
    matchEmailFromEmailPreferencesName(emailPreferencesName) {
        return this.pathTemplates.emailPreferencesPathTemplate.match(emailPreferencesName).email;
    }
    /**
     * Return a fully-qualified gbpAccount resource name string.
     *
     * @param {string} account
     * @param {string} gbp_account
     * @returns {string} Resource name string.
     */
    gbpAccountPath(account, gbpAccount) {
        return this.pathTemplates.gbpAccountPathTemplate.render({
            account: account,
            gbp_account: gbpAccount,
        });
    }
    /**
     * Parse the account from GbpAccount resource.
     *
     * @param {string} gbpAccountName
     *   A fully-qualified path representing GbpAccount resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromGbpAccountName(gbpAccountName) {
        return this.pathTemplates.gbpAccountPathTemplate.match(gbpAccountName).account;
    }
    /**
     * Parse the gbp_account from GbpAccount resource.
     *
     * @param {string} gbpAccountName
     *   A fully-qualified path representing GbpAccount resource.
     * @returns {string} A string representing the gbp_account.
     */
    matchGbpAccountFromGbpAccountName(gbpAccountName) {
        return this.pathTemplates.gbpAccountPathTemplate.match(gbpAccountName).gbp_account;
    }
    /**
     * Return a fully-qualified homepage resource name string.
     *
     * @param {string} account
     * @returns {string} Resource name string.
     */
    homepagePath(account) {
        return this.pathTemplates.homepagePathTemplate.render({
            account: account,
        });
    }
    /**
     * Parse the account from Homepage resource.
     *
     * @param {string} homepageName
     *   A fully-qualified path representing Homepage resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromHomepageName(homepageName) {
        return this.pathTemplates.homepagePathTemplate.match(homepageName).account;
    }
    /**
     * Return a fully-qualified lfpProvider resource name string.
     *
     * @param {string} account
     * @param {string} omnichannel_setting
     * @param {string} lfp_provider
     * @returns {string} Resource name string.
     */
    lfpProviderPath(account, omnichannelSetting, lfpProvider) {
        return this.pathTemplates.lfpProviderPathTemplate.render({
            account: account,
            omnichannel_setting: omnichannelSetting,
            lfp_provider: lfpProvider,
        });
    }
    /**
     * Parse the account from LfpProvider resource.
     *
     * @param {string} lfpProviderName
     *   A fully-qualified path representing LfpProvider resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromLfpProviderName(lfpProviderName) {
        return this.pathTemplates.lfpProviderPathTemplate.match(lfpProviderName).account;
    }
    /**
     * Parse the omnichannel_setting from LfpProvider resource.
     *
     * @param {string} lfpProviderName
     *   A fully-qualified path representing LfpProvider resource.
     * @returns {string} A string representing the omnichannel_setting.
     */
    matchOmnichannelSettingFromLfpProviderName(lfpProviderName) {
        return this.pathTemplates.lfpProviderPathTemplate.match(lfpProviderName).omnichannel_setting;
    }
    /**
     * Parse the lfp_provider from LfpProvider resource.
     *
     * @param {string} lfpProviderName
     *   A fully-qualified path representing LfpProvider resource.
     * @returns {string} A string representing the lfp_provider.
     */
    matchLfpProviderFromLfpProviderName(lfpProviderName) {
        return this.pathTemplates.lfpProviderPathTemplate.match(lfpProviderName).lfp_provider;
    }
    /**
     * Return a fully-qualified omnichannelSetting resource name string.
     *
     * @param {string} account
     * @param {string} omnichannel_setting
     * @returns {string} Resource name string.
     */
    omnichannelSettingPath(account, omnichannelSetting) {
        return this.pathTemplates.omnichannelSettingPathTemplate.render({
            account: account,
            omnichannel_setting: omnichannelSetting,
        });
    }
    /**
     * Parse the account from OmnichannelSetting resource.
     *
     * @param {string} omnichannelSettingName
     *   A fully-qualified path representing OmnichannelSetting resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromOmnichannelSettingName(omnichannelSettingName) {
        return this.pathTemplates.omnichannelSettingPathTemplate.match(omnichannelSettingName).account;
    }
    /**
     * Parse the omnichannel_setting from OmnichannelSetting resource.
     *
     * @param {string} omnichannelSettingName
     *   A fully-qualified path representing OmnichannelSetting resource.
     * @returns {string} A string representing the omnichannel_setting.
     */
    matchOmnichannelSettingFromOmnichannelSettingName(omnichannelSettingName) {
        return this.pathTemplates.omnichannelSettingPathTemplate.match(omnichannelSettingName).omnichannel_setting;
    }
    /**
     * Return a fully-qualified onlineReturnPolicy resource name string.
     *
     * @param {string} account
     * @param {string} return_policy
     * @returns {string} Resource name string.
     */
    onlineReturnPolicyPath(account, returnPolicy) {
        return this.pathTemplates.onlineReturnPolicyPathTemplate.render({
            account: account,
            return_policy: returnPolicy,
        });
    }
    /**
     * Parse the account from OnlineReturnPolicy resource.
     *
     * @param {string} onlineReturnPolicyName
     *   A fully-qualified path representing OnlineReturnPolicy resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromOnlineReturnPolicyName(onlineReturnPolicyName) {
        return this.pathTemplates.onlineReturnPolicyPathTemplate.match(onlineReturnPolicyName).account;
    }
    /**
     * Parse the return_policy from OnlineReturnPolicy resource.
     *
     * @param {string} onlineReturnPolicyName
     *   A fully-qualified path representing OnlineReturnPolicy resource.
     * @returns {string} A string representing the return_policy.
     */
    matchReturnPolicyFromOnlineReturnPolicyName(onlineReturnPolicyName) {
        return this.pathTemplates.onlineReturnPolicyPathTemplate.match(onlineReturnPolicyName).return_policy;
    }
    /**
     * Return a fully-qualified program resource name string.
     *
     * @param {string} account
     * @param {string} program
     * @returns {string} Resource name string.
     */
    programPath(account, program) {
        return this.pathTemplates.programPathTemplate.render({
            account: account,
            program: program,
        });
    }
    /**
     * Parse the account from Program resource.
     *
     * @param {string} programName
     *   A fully-qualified path representing Program resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromProgramName(programName) {
        return this.pathTemplates.programPathTemplate.match(programName).account;
    }
    /**
     * Parse the program from Program resource.
     *
     * @param {string} programName
     *   A fully-qualified path representing Program resource.
     * @returns {string} A string representing the program.
     */
    matchProgramFromProgramName(programName) {
        return this.pathTemplates.programPathTemplate.match(programName).program;
    }
    /**
     * Return a fully-qualified region resource name string.
     *
     * @param {string} account
     * @param {string} region
     * @returns {string} Resource name string.
     */
    regionPath(account, region) {
        return this.pathTemplates.regionPathTemplate.render({
            account: account,
            region: region,
        });
    }
    /**
     * Parse the account from Region resource.
     *
     * @param {string} regionName
     *   A fully-qualified path representing Region resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromRegionName(regionName) {
        return this.pathTemplates.regionPathTemplate.match(regionName).account;
    }
    /**
     * Parse the region from Region resource.
     *
     * @param {string} regionName
     *   A fully-qualified path representing Region resource.
     * @returns {string} A string representing the region.
     */
    matchRegionFromRegionName(regionName) {
        return this.pathTemplates.regionPathTemplate.match(regionName).region;
    }
    /**
     * Return a fully-qualified shippingSettings resource name string.
     *
     * @param {string} account
     * @returns {string} Resource name string.
     */
    shippingSettingsPath(account) {
        return this.pathTemplates.shippingSettingsPathTemplate.render({
            account: account,
        });
    }
    /**
     * Parse the account from ShippingSettings resource.
     *
     * @param {string} shippingSettingsName
     *   A fully-qualified path representing ShippingSettings resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromShippingSettingsName(shippingSettingsName) {
        return this.pathTemplates.shippingSettingsPathTemplate.match(shippingSettingsName).account;
    }
    /**
     * Return a fully-qualified termsOfService resource name string.
     *
     * @param {string} version
     * @returns {string} Resource name string.
     */
    termsOfServicePath(version) {
        return this.pathTemplates.termsOfServicePathTemplate.render({
            version: version,
        });
    }
    /**
     * Parse the version from TermsOfService resource.
     *
     * @param {string} termsOfServiceName
     *   A fully-qualified path representing TermsOfService resource.
     * @returns {string} A string representing the version.
     */
    matchVersionFromTermsOfServiceName(termsOfServiceName) {
        return this.pathTemplates.termsOfServicePathTemplate.match(termsOfServiceName).version;
    }
    /**
     * Return a fully-qualified termsOfServiceAgreementState resource name string.
     *
     * @param {string} account
     * @param {string} identifier
     * @returns {string} Resource name string.
     */
    termsOfServiceAgreementStatePath(account, identifier) {
        return this.pathTemplates.termsOfServiceAgreementStatePathTemplate.render({
            account: account,
            identifier: identifier,
        });
    }
    /**
     * Parse the account from TermsOfServiceAgreementState resource.
     *
     * @param {string} termsOfServiceAgreementStateName
     *   A fully-qualified path representing TermsOfServiceAgreementState resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromTermsOfServiceAgreementStateName(termsOfServiceAgreementStateName) {
        return this.pathTemplates.termsOfServiceAgreementStatePathTemplate.match(termsOfServiceAgreementStateName).account;
    }
    /**
     * Parse the identifier from TermsOfServiceAgreementState resource.
     *
     * @param {string} termsOfServiceAgreementStateName
     *   A fully-qualified path representing TermsOfServiceAgreementState resource.
     * @returns {string} A string representing the identifier.
     */
    matchIdentifierFromTermsOfServiceAgreementStateName(termsOfServiceAgreementStateName) {
        return this.pathTemplates.termsOfServiceAgreementStatePathTemplate.match(termsOfServiceAgreementStateName).identifier;
    }
    /**
     * Return a fully-qualified user resource name string.
     *
     * @param {string} account
     * @param {string} email
     * @returns {string} Resource name string.
     */
    userPath(account, email) {
        return this.pathTemplates.userPathTemplate.render({
            account: account,
            email: email,
        });
    }
    /**
     * Parse the account from User resource.
     *
     * @param {string} userName
     *   A fully-qualified path representing User resource.
     * @returns {string} A string representing the account.
     */
    matchAccountFromUserName(userName) {
        return this.pathTemplates.userPathTemplate.match(userName).account;
    }
    /**
     * Parse the email from User resource.
     *
     * @param {string} userName
     *   A fully-qualified path representing User resource.
     * @returns {string} A string representing the email.
     */
    matchEmailFromUserName(userName) {
        return this.pathTemplates.userPathTemplate.match(userName).email;
    }
    /**
     * Terminate the gRPC channel and close the client.
     *
     * The client will no longer be usable and all future behavior is undefined.
     * @returns {Promise} A promise that resolves when the client is closed.
     */
    close() {
        if (this.emailPreferencesServiceStub && !this._terminated) {
            return this.emailPreferencesServiceStub.then(stub => {
                this._log.info('ending gRPC channel');
                this._terminated = true;
                stub.close();
            });
        }
        return Promise.resolve();
    }
}
exports.EmailPreferencesServiceClient = EmailPreferencesServiceClient;
